fbfmaproom/environment.yml

```diff
@@ -3,15 +3,30 @@ channels:
   - conda-forge
   - defaults
 dependencies:
+
+    # rasterio build fails with python 3.10
+    # https://github.com/rasterio/rasterio/issues/2333
+  - python=3.9
   - cftime
   - dash
   - flask
   - numpy
   - opencv
-  - pandas
+
+    # Newer versions of Pandas do not appear to work correctly with queuepool.
+    # In particular, how that package forms SQL query objects needs to be updated. 
+    # Hopefully the solution here is to either send Igor a issue or to migrate to
+    # a different connection pooling library
```

Comment:
More precisely, new versions of `pandas.read_sql` and `pandas.read_sql_query` no longer accept a DBAPI object as the connection. Apparently it worked by coincidence, not by design. A change made last July [made it an error](https://github.com/pandas-dev/pandas/pull/42546/files#diff-9268174bfb15f08ef2267375665a85fecf201999902542f6fc9c0d3fadfb4553R742), but people [complained](https://github.com/pandas-dev/pandas/issues/45416), so they [changed it to a warning](https://github.com/pandas-dev/pandas/pull/45496/files#diff-9268174bfb15f08ef2267375665a85fecf201999902542f6fc9c0d3fadfb4553R747).

The solution appears to be to use psycopg2 via SQLAlchemy rather than directly. I think that's a good move regardless; it's the community standard.

Reply:
And once we have SQLAlchemy as a dependency, there's no cost to also using its QueuePool instead of queuepool, which should solve the problem of applications hanging when the database server restarts.